content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -0,0 +1,219 @@
+Gate](/docs/reference/command-line-tools-reference/feature-gates.md) to be
+enabled and kubernetes client and server version of at least 1.16.
+{{< /caution >}}
+
+The examples in this section demonstrate how ephemeral containers appear in
+the API. For a more practical example of troubleshooting, see the [Debug a
+Running Pod](/docs/tasks/debug-application-cluster/debug-running-pod.md) task.
+
+Ephemeral containers are created using the `ephemeralcontainers` subresource
+of pod, which can be demonstrated using `kubectl --raw`. First describe
+the ephemeral container to add as an `EphemeralContainers` list:
+
+```json
+{
+    "apiVersion": "v1",
+    "kind": "EphemeralContainers",
+    "metadata": {
+            "name": "example-pod"
+    },
+    "ephemeralContainers": [{
+        "command": [
+            "sh"
+        ],
+        "image": "busybox",
+        "imagePullPolicy": "IfNotPresent",
+        "name": "debugger",
+        "stdin": true,
+        "tty": true,
+        "terminationMessagePolicy": "File"
+    }]
+}
+```
+
+To update the ephemeral containers of the already running `example-pod`:
+
+```shell
+kubectl replace --raw /api/v1/namespaces/default/pods/example-pod/ephemeralcontainers  -f ec.json
+```
+
+This will return the new list of ephemeral containers:
+
+```json
+{
+   "kind":"EphemeralContainers",
+   "apiVersion":"v1",
+   "metadata":{
+      "name":"example-pod",
+      "namespace":"default",
+      "selfLink":"/api/v1/namespaces/default/pods/example-pod/ephemeralcontainers",
+      "uid":"a14a6d9b-62f2-4119-9d8e-e2ed6bc3a47c",
+      "resourceVersion":"15886",
+      "creationTimestamp":"2019-08-29T06:41:42Z"
+   },
+   "ephemeralContainers":[
+      {
+         "name":"debugger",
+         "image":"busybox",
+         "command":[
+            "sh"
+         ],
+         "resources":{
+
+         },
+         "terminationMessagePolicy":"File",
+         "imagePullPolicy":"IfNotPresent",
+         "stdin":true,
+         "tty":true
+      }
+   ]
+}
+```
+
+The ephemeral container now appears in a `kubectl describe`:
+
+```shell
+kubectl describe pod example-pod
+```
+
+```
+...
+Ephemeral Containers:
+  debugger:
+    Container ID:  docker://cf81908f149e7e9213d3c3644eda55c72efaff67652a2685c1146f0ce151e80f
+    Image:         busybox
+    Image ID:      docker-pullable://busybox@sha256:9f1003c480699be56815db0f8146ad2e22efea85129b5b5983d0e0fb52d9ab70
+    Port:          <none>
+    Host Port:     <none>
+    Command:
+      sh
+    State:          Running
+      Started:      Thu, 29 Aug 2019 06:42:21 +0000
+    Ready:          False
+    Restart Count:  0
+    Environment:    <none>
+    Mounts:         <none>
+...
+```
+
+And can be attached using `kubectl attach`:
+
+```shell
+kubectl attach -it example-pod -c debugger
+```
+
+When process namespace sharing is enabled, all pod processes are visible. For example:
+
+```shell
+% kubectl attach -it example-pod -c debugger
```

Comment:
nits:
1. Avoid including the prompt
2. Put the sample output in a separate block (and don't mark it as Bourne shell code)

Reply:
This is the separate block including the output (the command was in a previous block). I thought showing the two prompts (one outside the container and one inside), but I can remove the command from the output.